tck/src/test/java/org/eclipse/microprofile/jwt/tck/container/jaxrs/CookieTokenTest.java

```diff
@@ -0,0 +1,167 @@
+                        "HTTP_UNAUTHORIZED")
+    public void wrongCookieName() throws Exception {
+        String token = TokenUtils.generateTokenString("/Token1.json");
+
+        String uri = baseURL.toExternalForm() + "endp/echo";
+        WebTarget echoEndpointTarget = ClientBuilder.newClient()
+                                                    .target(uri)
+                                                    .queryParam("input", "hello");
+        Response response = echoEndpointTarget
+            .request(TEXT_PLAIN)
+            .cookie("Bearer", token)
+            .get();
+        Assert.assertEquals(response.getStatus(), HttpURLConnection.HTTP_UNAUTHORIZED);
+        String reply = response.readEntity(String.class);
+        System.out.printf("Reply: %s\n", reply);
+    }
+
+    @RunAsClient
+    @Test(groups = TEST_GROUP_JAXRS,
+          description = "Validate a request without empty token in a Cookie fails with HTTP_UNAUTHORIZED")
+    public void emptyCookie() throws Exception {
+        String token = TokenUtils.generateTokenString("/Token1.json");
+
+        String uri = baseURL.toExternalForm() + "endp/echo";
+        WebTarget echoEndpointTarget = ClientBuilder.newClient()
+                                                    .target(uri)
+                                                    .queryParam("input", "hello");
+        Response response = echoEndpointTarget
+            .request(TEXT_PLAIN)
+            .cookie("jwt", "")
+            .get();
+        Assert.assertEquals(response.getStatus(), HttpURLConnection.HTTP_UNAUTHORIZED);
+        String reply = response.readEntity(String.class);
+        System.out.printf("Reply: %s\n", reply);
+    }
+
+    @RunAsClient
+    @Test(groups = TEST_GROUP_JAXRS,
+          description = "Validate a request with valid token in Header but endpoints expects Cookie fails with " +
+                        "HTTP_UNAUTHORIZED")
+    public void ignoreHeaderIfCookieSet() throws Exception {
+        String token = TokenUtils.generateTokenString("/Token1.json");
+
+        String uri = baseURL.toExternalForm() + "endp/echo";
+        WebTarget echoEndpointTarget = ClientBuilder.newClient()
+                                                    .target(uri)
+                                                    .queryParam("input", "hello");
+        Response response = echoEndpointTarget
+            .request(TEXT_PLAIN)
+            .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+            .get();
+        Assert.assertEquals(response.getStatus(), HttpURLConnection.HTTP_UNAUTHORIZED);
+        String reply = response.readEntity(String.class);
+        System.out.printf("Reply: %s\n", reply);
+    }
+
+
+    @RunAsClient
+    @Test(groups = TEST_GROUP_JAXRS,
+          description = "Validate a request with expired token in a Cookie fails with HTTP_UNAUTHORIZED")
+    public void expiredCookie() throws Exception {
```

Comment:
Having this test does not harm, but the token validaton (exp, woron issuer, etc) is already well covered, we are just testing here that the alt delivery channels work; if you'd like to keep it it is fine :-)

Reply:
Yes, I know it was just to have a different test :)